pkg/controller/devicetainteviction/device_taint_eviction.go

```diff
@@ -0,0 +1,917 @@
+		AddFunc: func(obj any) {
+			claim, ok := obj.(*resourceapi.ResourceClaim)
+			if !ok {
+				logger.Error(nil, "Expected ResourceClaim", "actual", fmt.Sprintf("%T", obj))
+				return
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handleClaimChange(nil, claim)
+		},
+		UpdateFunc: func(oldObj, newObj any) {
+			oldClaim, ok := oldObj.(*resourceapi.ResourceClaim)
+			if !ok {
+				logger.Error(nil, "Expected ResourceClaim", "actual", fmt.Sprintf("%T", oldObj))
+				return
+			}
+			newClaim, ok := newObj.(*resourceapi.ResourceClaim)
+			if !ok {
+				logger.Error(nil, "Expected ResourceClaim", "actual", fmt.Sprintf("%T", newObj))
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handleClaimChange(oldClaim, newClaim)
+		},
+		DeleteFunc: func(obj any) {
+			if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+				obj = tombstone.Obj
+			}
+			claim, ok := obj.(*resourceapi.ResourceClaim)
+			if !ok {
+				logger.Error(nil, "Expected ResourceClaim", "actual", fmt.Sprintf("%T", obj))
+				return
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handleClaimChange(claim, nil)
+		},
+	})
+	defer func() {
+		_ = tc.claimInformer.Informer().RemoveEventHandler(claimHandler)
+	}()
+	tc.haveSynced = append(tc.haveSynced, claimHandler.HasSynced)
+
+	podHandler, _ := tc.podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj any) {
+			pod, ok := obj.(*v1.Pod)
+			if !ok {
+				logger.Error(nil, "Expected ResourcePod", "actual", fmt.Sprintf("%T", obj))
+				return
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handlePodChange(nil, pod)
+		},
+		UpdateFunc: func(oldObj, newObj any) {
+			oldPod, ok := oldObj.(*v1.Pod)
+			if !ok {
+				logger.Error(nil, "Expected Pod", "actual", fmt.Sprintf("%T", oldObj))
+				return
+			}
+			newPod, ok := newObj.(*v1.Pod)
+			if !ok {
+				logger.Error(nil, "Expected Pod", "actual", fmt.Sprintf("%T", newObj))
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handlePodChange(oldPod, newPod)
+		},
+		DeleteFunc: func(obj any) {
+			if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+				obj = tombstone.Obj
+			}
+			pod, ok := obj.(*v1.Pod)
+			if !ok {
+				logger.Error(nil, "Expected Pod", "actual", fmt.Sprintf("%T", obj))
+				return
+			}
+			mutex.Lock()
+			defer mutex.Unlock()
+			tc.handlePodChange(pod, nil)
+		},
+	})
+	defer func() {
+		_ = tc.podInformer.Informer().RemoveEventHandler(podHandler)
+	}()
+	tc.haveSynced = append(tc.haveSynced, podHandler.HasSynced)
+
+	opts := resourceslicetracker.Options{
+		EnableDeviceTaints: true,
+		SliceInformer:      tc.sliceInformer,
+		TaintInformer:      tc.taintInformer,
+		ClassInformer:      tc.classInformer,
+		KubeClient:         tc.client,
+	}
+	sliceTracker, err := resourceslicetracker.StartTracker(ctx, opts)
+	if err != nil {
+		logger.Info("Failed to initialize ResourceSlice patch tracker, stopping to evict pods", "err", err)
```

Comment:
Is this log entry equivalent?
```suggestion
		logger.Info("Failed to initialize ResourceSlice patch tracker; device taints leading to Pod eviction are now paused", "err", err)
```

I couldn't tell.

Reply:
Yes, applied (and "patch" removed - we no longer have patches).